_posts/en/newsletters/2019-02-05-newsletter.md

```diff
@@ -0,0 +1,283 @@
+    (about 10% of the time, it'd be within 3 weeks; but another 10% of
+    the time, it'd take more than a year).
+
+    Miners would have no control over whether or not their block
+    signaled for activation, but they would have control over whether
+    they published that block.  A miner who refused to publish his own
+    block if it signaled for activation would lose the income from that
+    block but would successfully prevent the fork from activating at
+    least until the next signaling block was produced (which could be
+    tomorrow or could be two years later).  This gives miners a real
+    chance to hold back a change but only by sacrificing real income.
+    The end of the talk suggests some variations on the method with
+    different tradeoffs.
+
+    The idea needs to be analyzed for possible problems, but it does
+    provide an interesting alternative to [BIP9][]-style miner-activated
+    soft forks (MASFs) and [BIP8][]-style user-activated soft forks (UASFs).
+
+    At the conclusion of his talk, this speaker also announced that the
+    next Scaling Bitcoin conference and EdgeDev++ training sessions will
+    be later in 2019 in Tel Aviv, Israel.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+and [libsecp256k1][libsecp256k1 repo].*
+
+- [Bitcoin Core #14929][] allows peers that your node automatically
+  disconnected for misbehavior (e.g. sending invalid data) to reconnect
+  to your node if you have unused incoming connection slots.  If your
+  slots fill up, the misbehaving nodes are disconnected to make room for
+  nodes without a history of problems (unless the misbehaving node helps
+  your node in some other way, such as by connecting to a part of the
+  Internet from which you don't have many other peers).  Previously,
+  Bitcoin Core banned the IP addresses of misbehaving peers for a period
+  of time (default of 1 day); this was easily circumvented by attackers
+  with multiple IP addresses.  This solution gives those peers a chance
+  to be useful but provides priority to potentially more helpful peers.
+  If you manually ban a peer, such as by using the `setban` RPC,
+  connections from that peer will still be rejected.
+
+- [Bitcoin Core #13926][] adds a new `bitcoin-wallet` tool to the
+  executables Bitcoin Core provides.  Without needing to use any RPCs,
+  this tool can currently create a new wallet file or display some basic
+  information about an existing wallet, such as whether the wallet is
+  encrypted, whether it uses an HD seed, how many transactions it
+  contains, and how many address book entries it has.  This helps people
+  who have a wallet file that hasn't been synced to the most recent chain
+  tip; they can do a quick inspection on the wallet to see if it's
+  interesting before they perform the lengthy rescan necessary to import
+  it.
```

Comment:
Nit: note that the tool is expected to be enhanced in future to offer more functionality, such as salvaging damaged wallet files or topping up keypools.